base/shell/explorer/trayntfy.cpp

```diff
@@ -308,17 +313,226 @@ class CIconWatcher
     }
 };
 
+class CBalloonQueue
+{
+public:
+    static const int TimerInterval = 2000;
+    static const int BalloonsTimerId = 1;
+    static const int MinTimeout = 10000;
+    static const int MaxTimeout = 30000;
+    static const int CooldownBetweenBalloons = 2000;
+
+private:
+    struct Info
+    {
+        InternalIconData * pSource;
+        WCHAR szInfo[256];
+        WCHAR szInfoTitle[64];
+        WPARAM uIcon;
+        UINT uTimeout;
+
+        Info(InternalIconData * source)
+        {
+            pSource = source;
+            StrNCpy(szInfo, source->szInfo, _countof(szInfo));
+            StrNCpy(szInfoTitle, source->szInfoTitle, _countof(szInfoTitle));
+            uIcon = source->dwInfoFlags & 3;
+            if (source->dwInfoFlags & NIIF_USER)
+                uIcon = reinterpret_cast<WPARAM>(source->hIcon);
+            uTimeout = source->uTimeout;
+        }
+    };
+
+    HWND m_hwndParent;
+
+    CTooltips * m_tooltips;
+
+    CAtlList<Info> m_queue;
+
+    CToolbar<InternalIconData> * m_toolbar;
+
+    InternalIconData * m_current;
+    bool m_closed;
+
+    int m_timer;
+
+public:
+    CBalloonQueue() :
+        m_hwndParent(NULL),
+        m_tooltips(NULL),
+        m_toolbar(NULL),
+        m_current(NULL),
+        m_closed(false),
+        m_timer(-1)
+    {
+    }
+
+    void Init(HWND hwndParent, CToolbar<InternalIconData> * toolbar, CTooltips * balloons)
```

Comment:
Check your `type *foo;` in the code. You have all three variants of spaces around `*` in a single file. 
Choose one for the consistency :wink: 
Same goes for `&`.

Reply:
There's no good way to solve this! 😱 Even if I make the ones I wrote **in this PR** consistent, there's other styles elsewhere in the file.
Basically: my inertia would normally be to put it next to the type, because the type SHOULD be "pointer to X" -> `X*`, but that's opposite to how C/C++ define the type declarations, where `X *a,b;" only makes `a` a pointer, so in theory it would be better to align it to the right, but that's ugly! So when I start fighting with that, I end up typing them with space on both sides, as a statement of my complaint for one of the most annoying design choices of C.

Reply:
I usually put `*` next to type unless I declare multiple pointers.

Reply:
Yeah so, FYI, I fixed all the other uses within the PR, but I left alone any other inconsistencies -- I'm not going to add noise to the diff at this point.